### PR TITLE
fix(pull_request)!::bug: Only `/WIP` command will work now automation is removed

### LINKED ISSUE #15: :bug: WIP label is added when title is changed but when WIP label is manually added title is not getting updated
## Description :-
When User is adding 🚧 emoji to title than `WIP` label is added by bot but when `WIP` label is added by user manually
than 🚧 emoji should be added in title

## To Reproduce :-

1. Go to 'Pull request'
2. Add label: WIP
3. Title is not updated

## Expected behavior :-
When `WIP` label is added than title should be updated

## Screenshots :-

Bug gif

### PR DESCRIPTION
**Pull request by aayushchugh**
# Description

Only `/WIP` command will work now automation is removed

Fixes #15 

# Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (postman)

### CODE REVIEW

File: automation/addLabelsOnPullRequest.automation.js [hunk 1]
```diff
@@ -65,103 +63,38 @@ exports.addMergedLabel = async context => {
...
-			title.includes('work in progress') ||
-			title.includes(':construction:')
+			title.includes("WIP") ||
+			title.includes("Work In Progress") ||
+			title.includes("work in progress") ||
+			title.includes(":construction:")
 		) {
 			const params = context.pullRequest({
-				title: `${title.replace(':construction:', '')}`,
+				title: `${title.replace(":construction:", "")}`,
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:90]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
        title: `${title.replace(":construction:", "")}`,
```

File: index.js [hunk 2]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
...
-				content: '+1',
+				content: "+1",
 			});
 
 			context.octokit.reactions.createForIssueComment(params);
 		}
 
 		switch (command) {
-			case '/label':
+			case "/label":
```
Comment (aayushchugh) [inline index.js:56]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      case "/label":
```

File: commands/wip.command.js [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
 	);
 
 	if (
-		title.includes('WIP') ||
-		title.includes('Work In Progress') ||
-		title.includes('work in progress') ||
-		title.includes(':construction:')
+		title.includes("WIP") ||
+		title.includes("Work In Progress") ||
+		title.includes("work in progress") ||
```
Comment (aayushchugh) [inline commands/wip.command.js:18]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    title.includes("work in progress") ||
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 4]
```diff
@@ -170,36 +103,36 @@ exports.changesRequestLabel = async context => {
...
-		label => label.name === ':white_check_mark: Approved'
+		label => label.name === ":white_check_mark: Approved"
 	);
 
 	if (
 		changesRequestedReviews.length > 0 &&
 		foundChangedRequestedLabel.length === 0
 	) {
-		addLabel([':warning: Changes requested'], context, 'AA2626');
+		addLabel([":warning: Changes requested"], context, "AA2626");
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:125]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    addLabel([":warning: Changes requested"], context, "AA2626");
```

File: index.js [hunk 5]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
...
-	app.on('pull_request.closed', addMergedLabel);
-
-	app.on(
-		['pull_request.edited', 'pull_request.labeled'],
-		pullRequestWIPLabelAutomation
-	);
-
-	app.on('issues.closed', addLabelToIssueOnClose);
+	app.on("pull_request_review.submitted", addApprovedLabel);
+	app.on("pull_request_review.submitted", changesRequestLabel);
```
Comment (aayushchugh) [inline index.js:36]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
  app.on("pull_request_review.submitted", changesRequestLabel);
```

File: commands/wip.command.js [hunk 6]
```diff
@@ -1,37 +1,58 @@
...
+		label => label.name === ":mag: Ready for Review"
 	);
 
 	if (
-		title.includes('WIP') ||
-		title.includes('Work In Progress') ||
-		title.includes('work in progress') ||
-		title.includes(':construction:')
+		title.includes("WIP") ||
+		title.includes("Work In Progress") ||
```
Comment (aayushchugh) [inline commands/wip.command.js:17]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    title.includes("Work In Progress") ||
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 7]
```diff
@@ -65,103 +63,38 @@ exports.addMergedLabel = async context => {
...
 		}
 		console.log(title);
 
 		if (
-			title.includes('WIP') ||
-			title.includes('Work In Progress') ||
-			title.includes('work in progress') ||
-			title.includes(':construction:')
+			title.includes("WIP") ||
+			title.includes("Work In Progress") ||
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:85]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      title.includes("Work In Progress") ||
```

File: commands/wip.command.js [hunk 8]
```diff
@@ -1,37 +1,58 @@
...
 		context.octokit.issues.update(params);
 	}
 
 	if (wipLabel) {
-		return removeLabel(':construction: WIP', context);
+		if (!foundReadyForReviewLabel) {
+			addLabel([":mag: Ready for Review"], context);
+		}
+
+		return removeLabel(":construction: WIP", context);
```
Comment (aayushchugh) [inline commands/wip.command.js:33]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    return removeLabel(":construction: WIP", context);
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 9]
```diff
@@ -18,42 +18,40 @@ exports.addApprovedLabel = async context => {
...
 	if (approvedReviews.length > 0) {
 		const issueLabels = await listIssueLabels(context);
 
 		const foundReviewLabel = issueLabels.data.filter(
-			label => label.name === ':mag: Ready for Review'
+			label => label.name === ":mag: Ready for Review"
 		);
 
-		addLabel([':white_check_mark: Approved'], context);
+		if (approvedLabel.length === 0) {
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:49]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    if (approvedLabel.length === 0) {
```

File: commands/wip.command.js [hunk 10]
```diff
@@ -1,37 +1,58 @@
...
-	addLabel([':construction: WIP'], context, '383214');
-	const params = context.issue({
-		title: `:construction: ${context.payload.issue.title}`,
-	});
+	if (!wipLabel) {
+		if (foundReadyForReviewLabel) {
+			removeLabel(":mag: Ready for Review", context);
+		}
+
+		addLabel([":construction: WIP"], context, "383214");
```
Comment (aayushchugh) [inline commands/wip.command.js:41]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    addLabel([":construction: WIP"], context, "383214");
```

File: index.js [hunk 11]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
...
+	app.on("pull_request.closed", addMergedLabel);
+	app.on("issues.closed", addLabelToIssueOnClose);
```
Comment (aayushchugh) [inline index.js:38]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
  app.on("issues.closed", addLabelToIssueOnClose);
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 12]
```diff
@@ -65,103 +63,38 @@ exports.addMergedLabel = async context => {
...
-			label => label.name === ':white_check_mark: Approved'
+			label => label.name === ":white_check_mark: Approved"
 		);
 		const foundWIPLabel = issueLabels.data.filter(
-			label => label.name === ':construction: WIP'
+			label => label.name === ":construction: WIP"
 		);
 
-		addLabel([':sparkles: Merged:'], context);
+		addLabel([":sparkles: Merged:"], context);
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:72]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    addLabel([":sparkles: Merged:"], context);
```

File: commands/wip.command.js [hunk 13]
```diff
@@ -1,37 +1,58 @@
...
-		title.includes('work in progress') ||
-		title.includes(':construction:')
+		title.includes("WIP") ||
+		title.includes("Work In Progress") ||
+		title.includes("work in progress") ||
+		title.includes(":construction:")
 	) {
 		const params = context.issue({
-			title: `${context.payload.issue.title.replace(':construction:', '')}`,
+			title: `${context.payload.issue.title.replace(":construction:", "")}`,
```
Comment (aayushchugh) [inline commands/wip.command.js:22]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      title: `${context.payload.issue.title.replace(":construction:", "")}`,
```

File: commands/wip.command.js [hunk 14]
```diff
@@ -1,37 +1,58 @@
...
+		if (foundReadyForReviewLabel) {
+			removeLabel(":mag: Ready for Review", context);
+		}
+
+		addLabel([":construction: WIP"], context, "383214");
+	}
 
-	context.octokit.issues.update(params);
+	if (
+		!title.includes("WIP") ||
```
Comment (aayushchugh) [inline commands/wip.command.js:45]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    !title.includes("WIP") ||
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 15]
```diff
@@ -65,103 +63,38 @@ exports.addMergedLabel = async context => {
...
+			removeLabel([":construction: WIP"], context);
 		}
 		console.log(title);
 
 		if (
-			title.includes('WIP') ||
-			title.includes('Work In Progress') ||
-			title.includes('work in progress') ||
-			title.includes(':construction:')
+			title.includes("WIP") ||
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:84]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      title.includes("WIP") ||
```

File: index.js [hunk 16]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
...
 			context.octokit.reactions.createForIssueComment(params);
 		}
 
 		switch (command) {
-			case '/label':
+			case "/label":
 				labelCommand(context, args);
 				break;
-			case '/close':
+			case "/close":
```
Comment (aayushchugh) [inline index.js:59]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      case "/close":
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 17]
```diff
@@ -65,103 +63,38 @@ exports.addMergedLabel = async context => {
...
-			label => label.name === ':construction: WIP'
+			label => label.name === ":construction: WIP"
 		);
 
-		addLabel([':sparkles: Merged:'], context);
+		addLabel([":sparkles: Merged:"], context);
 
 		if (foundApproveLabel.length > 0) {
-			removeLabel([':white_check_mark: Approved'], context);
+			removeLabel([":white_check_mark: Approved"], context);
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:75]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      removeLabel([":white_check_mark: Approved"], context);
```

File: index.js [hunk 18]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
  */
 module.exports = app => {
 	/* --------------------------------- ANCHOR Automation --------------------------------- */
-	app.on('pull_request.opened', addReadyForReviewLabel);
+	app.on("pull_request.opened", addReadyForReviewLabel);
```
Comment (aayushchugh) [inline index.js:33]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
  app.on("pull_request.opened", addReadyForReviewLabel);
```

File: commands/wip.command.js [hunk 19]
```diff
@@ -1,37 +1,58 @@
...
+		}
+
+		return removeLabel(":construction: WIP", context);
 	}
 
-	addLabel([':construction: WIP'], context, '383214');
-	const params = context.issue({
-		title: `:construction: ${context.payload.issue.title}`,
-	});
+	if (!wipLabel) {
```
Comment (aayushchugh) [inline commands/wip.command.js:36]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
  if (!wipLabel) {
```

File: index.js [hunk 20]
```diff
@@ -29,58 +30,51 @@ const availableCommandsMessage = `Available commands are:-
...
 
 	/* --------------------------------- ANCHOR Issue commands --------------------------------- */
-	app.on('issue_comment.created', async context => {
+	app.on("issue_comment.created", async context => {
 		const { body } = context.payload.comment;
 		const { command, args } = getCommandAndArgs(body);
 		const commentId = context.payload.comment.id;
 
-		if (command[0] === '/' && !context.isBot) {
+		if (command[0] === "/" && !context.isBot) {
```
Comment (aayushchugh) [inline index.js:46]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    if (command[0] === "/" && !context.isBot) {
```

File: commands/wip.command.js [hunk 21]
```diff
@@ -1,37 +1,58 @@
...
+		if (foundReadyForReviewLabel) {
+			removeLabel(":mag: Ready for Review", context);
```
Comment (aayushchugh) [inline commands/wip.command.js:38]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      removeLabel(":mag: Ready for Review", context);
```

File: commands/wip.command.js [hunk 22]
```diff
@@ -1,37 +1,58 @@
...
+			title: `${context.payload.issue.title.replace(":construction:", "")}`,
 		});
 
 		context.octokit.issues.update(params);
 	}
 
 	if (wipLabel) {
-		return removeLabel(':construction: WIP', context);
+		if (!foundReadyForReviewLabel) {
+			addLabel([":mag: Ready for Review"], context);
```
Comment (aayushchugh) [inline commands/wip.command.js:30]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
      addLabel([":mag: Ready for Review"], context);
```

File: commands/wip.command.js [hunk 23]
```diff
@@ -1,37 +1,58 @@
...
 const WIPCommand = async context => {
 	const { title } = context.payload.issue || context.payload.pull_request;
 	const labelsFromIssues = await listIssueLabels(context);
 
 	const wipLabel = labelsFromIssues.data.find(
-		label => label.name === ':construction: WIP'
+		label => label.name === ":construction: WIP"
+	);
+	const foundReadyForReviewLabel = labelsFromIssues.data.find(
+		label => label.name === ":mag: Ready for Review"
```
Comment (aayushchugh) [inline commands/wip.command.js:12]: Codacy has a fix for the issue: [Replace `↹↹label` with `····(label)`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060852)
```suggestion
    (label) => label.name === ":mag: Ready for Review"
```

**Review by Arpitchugh — APPROVED** [review]

**Pull request merged**